src/analyses/variable-sensitivity/abstract_enviroment.cpp

```diff
@@ -0,0 +1,585 @@
+        }
+      },
+      {
+        ID_dereference, [&](
+          const abstract_object_pointert lhs_object,
+          std::stack<exprt> stack,
+          abstract_object_pointert rhs_object)
+        {
+          sharing_ptrt<pointer_abstract_objectt> pointer_abstract_object=
+            std::dynamic_pointer_cast<pointer_abstract_objectt>(lhs_object);
+          sharing_ptrt<pointer_abstract_objectt> modified_pointer=
+            pointer_abstract_object->write_dereference(
+              *this, stactions, rhs_object, false);
+          return modified_pointer;
+        }
+      }
+    };
+
+    // We added something to the stack that we couldn't deal with
+    assert(handlers.find(next_expr.id())!=handlers.end());
+    assert(value);
+
+    // The symbol is not in the map - it is therefore top
+    abstract_object_pointert symbol_object;
+    if(map.find(symbol_expr)==map.end())
+    {
+      symbol_object=abstract_object_factory(symbol_expr.type(), true, false);
+    }
+    else
+    {
+      symbol_object=map[symbol_expr];
+    }
+    final_value=handlers[next_expr.id()](symbol_object, stactions, value);
+  }
+  else
+  {
+    // We can assign the AO directly to the symbol
+    final_value=value;
+  }
+
+  // Write the value for the root symbol back into the map
+  if (final_value->is_top())
+  {
+    map.erase(symbol_expr);
+
+  }
+  else
+  {
+    map[symbol_expr]=final_value;
+  }
+  return true;
+}
+
+/*******************************************************************\
+
+Function: abstract_environmentt::assume
+
+  Inputs:
+   expr - the expression inside the assume
+
+ Outputs: ?
+
+ Purpose: ?
+
+\*******************************************************************/
+
+bool abstract_environmentt::assume(const exprt &expr, const namespacet &ns)
+{
+  abstract_object_pointert res = eval(expr, ns);
+  std::string not_implemented_string=__func__;
+  not_implemented_string.append(" not implemented");
+  throw not_implemented_string;
+  // Need abstract_booleant
+#if 0
+  abstract_booleant *b = dynamic_cast<abstract_booleant>(res);
+
+  assert(b != NULL);
+
+  if (b->to_constant().is_false())
+  {
+    make_bottom();
+    return true;
+  }
+  else
+    return false;
+#endif
+}
+
+
+/*******************************************************************\
+
+Function: abstract_environmentt::abstract_object_factory
+
+  Inputs:
+   type - the type of the object whose state should be tracked
+   top - does the type of the object start as top
+
+ Outputs: The abstract object that has been created
+
+ Purpose: Look at the configuration for the sensitivity and create an
+          appropriate abstract_object
+
+\*******************************************************************/
+
+abstract_object_pointert abstract_environmentt::abstract_object_factory(
+  const typet type, bool top, bool bottom) const
+{
+  // TODO (tkiley): Here we should look at some config file
+  if(type.id()==ID_signedbv || type.id()==ID_bool || type.id()==ID_c_bool)
```

Comment:
What do those types have in common, and why are other types not part of this disjunction? (Not all of them are bitvector types, and it isn't all bitvector types either.)